--- mod/session.py	2021-02-05 20:40:28.803144031 +0000
+++ mod/session.py	2021-04-03 08:25:57.159223345 +0100
@@ -24,7 +24,6 @@
 from mod import safe_json_load, TextFileFlusher
 from mod.bank import get_last_bank_and_pedalboard
 from mod.development import FakeHost, FakeHMI
-from mod.hmi import HMI
 from mod.recorder import Recorder, Player
 from mod.screenshot import ScreenshotGenerator
 from mod.settings import (LOG,
@@ -81,12 +80,6 @@
         # Try to open real HMI
         hmiOpened = False
 
-        if not DEV_HMI:
-            self.hmi  = HMI(HMI_SERIAL_PORT, HMI_BAUD_RATE, HMI_TIMEOUT, self.hmi_initialized_cb, self.hmi_reinit_cb)
-            hmiOpened = self.hmi.sp is not None
-
-        #print("Using HMI =>", hmiOpened)
-
         if not hmiOpened:
             self.hmi = FakeHMI(self.hmi_initialized_cb)
             print("Using FakeHMI =>", self.hmi)
