--- mod/webserver.py
+++ mod/webserver.py
@@ -1751,7 +1751,6 @@ class TemplateHandler(TimelessRequestHandler):
             'version': self.get_argument('v'),
             'bin_compat': hwdesc.get('bin-compat', "Unknown"),
             'codec_truebypass': 'true' if hwdesc.get('codec_truebypass', False) else 'false',
-            'factory_pedalboards': hwdesc.get('factory_pedalboards', False),
             'platform': hwdesc.get('platform', "Unknown"),
             'addressing_pages': int(hwdesc.get('addressing_pages', 0)),
             'lv2_plugin_dir': LV2_PLUGIN_DIR,
@@ -2170,6 +2169,9 @@ class FilesList(JsonRequestHandler):
         elif filetype == "aidadspmodel":
             return ("Aida DSP Models", (".json",))
 
+        elif filetype == "nammodel":
+            return ("Amp Models", (".nam",))
+
         else:
             return (None, ())
 
