--- mod/settings.py
+++ mod/settings.py
@@ -43,7 +43,7 @@ else:
 
 DATA_DIR = os.environ.get('MOD_DATA_DIR', os.path.expanduser('~/data'))
 CACHE_DIR = os.path.join(DATA_DIR, '.cache')
-USER_FILES_DIR = os.environ.get('MOD_USER_FILES_DIR', '/data/user-files')
+USER_FILES_DIR = os.environ.get('MOD_USER_FILES_DIR', '/usr/share/mod/default-user-files')
 KEYS_PATH = os.environ.get('MOD_KEYS_PATH', join(DATA_DIR, 'keys'))
 FAVORITES_JSON_FILE = os.environ.get('MOD_FAVORITES_JSON', join(DATA_DIR, 'favorites.json'))
 LAST_STATE_JSON_FILE = os.environ.get('MOD_LAST_STATE_JSON', join(DATA_DIR, 'last.json'))
